### ISSUE TITLE
Adding an override with a server block causes duplicate warnings to be thrown

### ISSUE THREAD

**Issue by abea** [punkave/mechanic#15]
In `top` I have this on the `otherplace` mechanic overrides directory:
```
server {
  listen *:80;
  # Redirect old blog traffic to new blog section.
  server_name www.place.com place.com;

  location / {
    rewrite ^(.*)$ http://otherplace.com/first-place$1;
  }
}
```

Since `otherplace` has `--https=true` set, it outputs `top` above both the 80 and 443 server blocks. This throws the warning: 
```
nginx: [warn] conflicting server name "place.com" on 0.0.0.0:80, ignored
nginx: [warn] conflicting server name "place.com" on 0.0.0.0:80, ignored
```

There should be a way to have the override only printed once or use a port variable in the override (though that'd cause difficulty by requiring a cert on the redirected URL).

**Comment by boutell** [comment]
Agreed, I can't think of a situation where "top" being included twice is useful given that it's not inside any kind of block.

For this particular project, since what was wanted from "top" was essentially a completely separate (if trivial) site, we moved it to /etc/nginx/conf.d. But mechanic should still handle this right.
